instrumentation/scala-2.13.0/src/test/scala/com/nr/agent/instrumentation/scala/baseline/InstrumentedSegmentNestedAsynchronousMultipleThreadTest.scala

```diff
@@ -0,0 +1,107 @@
+@InstrumentationTestConfig(includePrefixes = Array("scala.concurrent.impl"), configName = "scala_futures_as_segments.yml")
+class InstrumentedSegmentNestedAsynchronousMultipleThreadTest {
+
+  val threadPoolOne: ExecutionContextExecutor = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(10))
+  val threadPoolTwo: ExecutionContextExecutor = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(10))
+  val threadPoolThree: ExecutionContextExecutor = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(10))
+
+  @Test
+  def oneNestedTransaction(): Unit = {
+    //Given
+    implicit val introspector: Introspector = InstrumentationTestRunner.getIntrospector
+
+    //When
+    val result = getOneResult
+
+    //Then
+    introspector.getFinishedTransactionCount(TimeUnit.SECONDS.toMillis(10))
+
+    val traces = getTraces()
+    val segments = getSegments(traces)
+
+    Assert.assertEquals("Result", 1, Await.result(result, 2.seconds))
+    Assert.assertEquals("Transactions", 1, introspector.getTransactionNames.size)
+    Assert.assertEquals("Traces", 1, traces.size)
+    Assert.assertEquals("Segments", 4, segments.size)
+  }
+
+  @Test
+  def twoNestedTransactions(): Unit = {
+    //Given
+    implicit val introspector: Introspector = InstrumentationTestRunner.getIntrospector
+
+    //When
+    val result = getTwoResults
+
+    //Then
+    introspector.getFinishedTransactionCount(TimeUnit.SECONDS.toMillis(10))
+
+    val traces = getTraces()
+    val segments = getSegments(traces)
+
+    Assert.assertEquals("Result", 3, Await.result(result, 2.seconds))
+    Assert.assertEquals("Transactions", 1, introspector.getTransactionNames.size)
+    Assert.assertEquals("Traces", 1, traces.size)
+    Assert.assertEquals("Segments", 7, segments.size)
+  }
+
+  @Test
+  def threeNestedTransactions(): Unit = {
+    //Given
+    implicit val introspector: Introspector = InstrumentationTestRunner.getIntrospector
+
+    //When
+    val result = getThreeResults
+
+    //Then
+    introspector.getFinishedTransactionCount(TimeUnit.SECONDS.toMillis(10))
+
+    val traces = getTraces()
+    val segments = getSegments(traces)
+
+    Assert.assertEquals("Result", 6, Await.result(result, 2.seconds))
+    Assert.assertEquals("Transactions", 1, introspector.getTransactionNames.size)
+    Assert.assertEquals("Traces", 1, traces.size)
+    Assert.assertEquals("Segments", 10, segments.size)
+  }
+
+  @Trace(dispatcher = true)
+  def getOneResult: Future[Int] = Future.reduceLeft(Seq(getFirstNumber))(_ + _)(threadPoolOne)
+
+  @Trace(dispatcher = true)
+  def getTwoResults: Future[Int] = Future.reduceLeft(Seq(getFirstNumber, getSecondNumber))(_ + _)(threadPoolTwo)
+
+  @Trace(dispatcher = true)
+  def getThreeResults: Future[Int] = Future.reduceLeft(Seq(getFirstNumber, getSecondNumber, getThirdNumber))(_ + _)(threadPoolThree)
+
+  @Trace
+  def getFirstNumber: Future[Int] = Future{
+    println(s"${Thread.currentThread.getName}: getFirstNumber")
```

Comment:
These println calls look like were used for debugging the tests.
But I think it is going to be just noise when the tests run normally.

Reply:
Removed println calls